api/src/main/scala/ai/chronon/api/Extensions.scala

```diff
@@ -631,6 +642,17 @@ object Extensions {
       }
     }
 
+    def getPartitionColumn: Option[String] = {
+      val partitionColumns = groupBy.sources.toScala.flatMap(_.partitionColumnOpt).distinct
+      if (partitionColumns.isEmpty) {
+        None
+      } else if (partitionColumns.length == 1) {
+        partitionColumns.headOption
+      } else {
+        throw new IllegalArgumentException(
+          s"Expect all queries from a given group-by to have the same partition column. All sources should have identical schemas and same partition column name. Found distinct partition columns $partitionColumns for group-by ${groupBy.metaData.name}")
+      }
```

Comment:
nit: could be simplified like so

```suggestion  
      val partitionColumns = groupBy.sources.toScala.flatMap(_.partitionColumnOpt).distinct
      
      require(partitionColumns.length < 2, s"Expect all queries from a given group-by to have the same partition column. All sources should have identical schemas and same partition column name. Found distinct partition columns $partitionColumns for group-by ${groupBy.metaData.name}")
      
      partitionColumns.headOption
    }
```